File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        counter={}
        heap=[]
        res=[]
        for n in nums:
            counter[n]=counter.get(n,0)+1
        
        for key,val in counter.items():
            heapq.heappush(heap,(-val,key))

        while len(res)<k:
            res.append(heapq.heappop(heap)[1])
        
        return res

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        HashMap<Integer, Integer> map = new HashMap<>();
        for(int i: nums){
            map.put(i,map.getOrDefault(i,0)+1);
        }
        List<Map.Entry<Integer, Integer>> entries = new ArrayList<>(map.entrySet());
        entries.sort((e1,e2)-> {
            return e2.getValue() - e1.getValue();
        });
        int[] arr = new int[k];
        for(int j=0;j<k;j++){
            arr[j]=entries.get(j).getKey();
        }
        return arr;
    }
}